### DISCUSSION TITLE
plugins: default directory for plugins

### DISCUSSION THREAD

**Discussion by kostya-zero** [wezterm/wezterm#4125] [Q&A]
Hi, everyone. I am developing a plugin manager for WezTerm. Where WezTerm looks for plugins by default?

**Comment by axelhj** [comment]
Hi!

I was kind of wondering the same; I have some scripts to scroll into the scrollback in cmd & resize fonts with scrollwheel at the same time, etc. and would like to make a plugin out of those scripts to shorten my config-file.

Supposedly it is possible to make some convention-based plugin-folder; eg any folder in `$XDG_DATA_HOME/wezterm/plugins/<my_plugin_name>/init.lua` is considered a plugin and required by a script that `.wezterm.lua` calls at startup. I'm considering making a plugin managementscript myself, basically inspired by Neovim which just clones github paths into a predefined base-folder.

Any thoughts?

EDIT: Apparently $XDG_CONFIG_HOME paths are already automatically require-able from the wezterm config folder which is used by existing plugins https://github.com/sei40kr/wez-status-generator?tab=readme-ov-file#installation.

Regards
Axel

**Reply by axelhj** [reply]
Update: There is a plugin system; https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd

I'm seeing now that WezTerm has a plugin system. It's mentioned in the [doc-site](https://wezfurlong.org/wezterm/index.html), though it doesn't have its own article. It is in the changelog. Searching "plugin" didn't yield any references though. I'm thinking about filing a bug towards the doc-site or trying to contribute something.
